subprojects/test-runner/device-provider/src/main/kotlin/com/avito/android/runner/devices/internal/KubernetesDevicesProvider.kt

```diff
@@ -31,8 +32,10 @@ internal class KubernetesDevicesProvider(
             adbDeviceFactory.create(
                 coordinate = coordinate,
                 adbDeviceParams = adbDeviceParams
-            )
+            ).onFailure { releaseDevice(coordinate) }
         }
+            .filter { it.isSuccess() }
+            .map { it.getOrThrow() }
```

Comment:
while i understand what is going on here, still it's hard to read; maybe good place for extension fun...